data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/MongoQueryBuilder.java

```diff
@@ -176,7 +176,7 @@ public final class MongoQueryBuilder implements QueryBuilder {
         addCriterionHandler(QueryModel.In.class, (context, obj, criterion) -> {
             PersistentPropertyPath propertyPath = context.getRequiredProperty(criterion);
             Object value = criterion.getValue();
-            String criterionPropertyName = getPropertyPersistName(context.getRequiredProperty(criterion).getProperty());
+            String criterionPropertyName = getCriterionPropertyName(criterion.getProperty(), context);
```

Comment:
I have changed this in #1907 wanting to reuse but was not correct to use this method here so I am reverting it.